src/video_core/renderer_vulkan/liverpool_to_vk.cpp

```diff
@@ -665,16 +665,36 @@ std::span<const SurfaceFormatInfo> SurfaceFormats() {
     return formats;
 }
 
+// Table 8.13 Data and Image Formats [Sea Islands Series Instruction Set Architecture]
+static const size_t amd_gpu_data_format_bit_size = 6; // All values are under 64
+static const size_t amd_gpu_number_format_bit_size = 4; // All values are under 16
+
+static size_t GetSurfaceFormatTableIndex(AmdGpu::DataFormat data_format, AmdGpu::NumberFormat num_format) {
+    DEBUG_ASSERT(data_format < 1 << amd_gpu_data_format_bit_size);
+    DEBUG_ASSERT(num_format < 1 << amd_gpu_number_format_bit_size);
+    size_t result = static_cast<size_t>(num_format) | (static_cast<size_t>(data_format) << amd_gpu_number_format_bit_size);
+    return result;
+}
+
+static auto surface_format_table = []() constexpr {
```

Comment:
Why do we need a separate table, can we not just make the existing table above indexed like this?

Reply:
Actually I guess this is fine, probably easier to define and iterate the format table with it sequential...

Reply:
I didn't want to mess with existing code. But I'll look into that.

Reply:
If something like a unordered_map where we can both lookup by this index and easily iterate the values is performant enough, that would be ideal for a single table. But otherwise we can probably just leave it as is for now, at least the lookup table is generated from it.

Reply:
Looks like std::unordered_map isn't something that can be spit out by a constexpr and live in rodata. It would have to be initialized at runtime. If it is built at runtime, it's just going to be two copies of the table anyway. Just one would live in the heap, and the other would live in rodata, instead of both living in rodata as the PR does as of my last change.

There are libraries that can do it at compile time (and also give perfect hashing), but I don't know if anything like it is currently in the project. One example: https://github.com/mapbox/eternal